example/app.py

```diff
@@ -1,53 +1,141 @@
+import asyncio
+import datetime
+import logging
 import os
-import time
 
-from realtime.channel import Channel
-from realtime.connection import Socket
+from realtime import AsyncRealtimeChannel, AsyncRealtimeClient
 
-
-def broadcast_callback(payload):
-    print("broadcast: ", payload)
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s - %(message)s", level=logging.INFO
+)
 
 
 def presence_callback(payload):
     print("presence: ", payload)
 
 
-def postgres_changes_callback(payload):
-    print("postgres_changes: ", payload)
+def postgres_changes_callback(payload, *args):
+    print("*: ", payload)
+
+
+def postgres_changes_insert_callback(payload, *args):
+    print("INSERT: ", payload)
+
+
+def postgres_changes_delete_callback(payload, *args):
+    print("DELETE: ", payload)
+
+
+def postgres_changes_update_callback(payload, *args):
+    print("UPDATE: ", payload)
 
 
 async def realtime(payload):
     print("async realtime ", payload)
 
 
-if __name__ == "__main__":
-    ID = os.getenv("SUPABASE_ID")
-    URL = f"https://{ID}.supabase.co"
-    JWT = os.getenv("API_KEY")
+async def test_broadcast_events(socket: AsyncRealtimeClient):
+    await socket.connect()
+
+    channel = socket.channel(
+        "test-broadcast", params={"config": {"broadcast": {"self": True}}}
+    )
+    received_events = []
+
+    def broadcast_callback(payload, *args):
+        print("broadcast: ", payload)
+        received_events.append(payload)
+
+    await channel.on_broadcast("test-event", callback=broadcast_callback).subscribe()
+
+    await asyncio.sleep(1)
+
+    # Send 3 broadcast events
+    for i in range(3):
+        await channel.send_broadcast("test-event", {"message": f"Event {i+1}"})
+
+    # Wait a short time to ensure all events are processed
+    await asyncio.sleep(1)
+
+    assert len(received_events) == 3
+    assert received_events[0]["payload"]["message"] == "Event 1"
+    assert received_events[1]["payload"]["message"] == "Event 2"
+    assert received_events[2]["payload"]["message"] == "Event 3"
+
+
+async def test_postgres_changes(socket: AsyncRealtimeClient):
+    await socket.connect()
+
+    await socket.set_auth(
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJhdXRoZW50aWNhdGVkIiwiZXhwIjoxNzIzNzUzNjI3LCJpYXQiOjE3MjM3NTAwMjcsImlzcyI6Imh0dHA6Ly8xMjcuMC4wLjE6NTQzMjEvYXV0aC92MSIsInN1YiI6ImYwNDBmNmY4LTA2YTUtNDA1My04YzQzLTcyMjBlZWMzZDMyZSIsImVtYWlsIjoidGVzdEBtYWlsLmNvbSIsInBob25lIjoiIiwiYXBwX21ldGFkYXRhIjp7InByb3ZpZGVyIjoiZW1haWwiLCJwcm92aWRlcnMiOlsiZW1haWwiXX0sInVzZXJfbWV0YWRhdGEiOnt9LCJyb2xlIjoiYXV0aGVudGljYXRlZCIsImFhbCI6ImFhbDEiLCJhbXIiOlt7Im1ldGhvZCI6InBhc3N3b3JkIiwidGltZXN0YW1wIjoxNzIzNzUwMDI3fV0sInNlc3Npb25faWQiOiJkMzA2OTc0Ni1iOTE5LTRkOTgtOWQ2Mi01MmRhYTZmMmIyZTgiLCJpc19hbm9ueW1vdXMiOmZhbHNlfQ.k-lLY0ogNZKEsvZNXP5xgi4hgBcRh0IbQtkc-Y_QXcs"
```

Comment:
Probably doesn't affect the test but can consider setting a later expiry like below as the token above will expire today 

```suggestion    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJhdXRoZW50aWNhdGVkIiwiZXhwIjoyMDIzNzUzNjI3LCJpYXQiOjE3MjM3NTAwMjcsImlzcyI6Imh0dHA6Ly8xMjcuMC4wLjE6NTQzMjEvYXV0aC92MSIsInN1YiI6ImYwNDBmNmY4LTA2YTUtNDA1My04YzQzLTcyMjBlZWMzZDMyZSIsImVtYWlsIjoidGVzdEBtYWlsLmNvbSIsInBob25lIjoiIiwiYXBwX21ldGFkYXRhIjp7InByb3ZpZGVyIjoiZW1haWwiLCJwcm92aWRlcnMiOlsiZW1haWwiXX0sInVzZXJfbWV0YWRhdGEiOnt9LCJyb2xlIjoiYXV0aGVudGljYXRlZCIsImFhbCI6ImFhbDEiLCJhbXIiOlt7Im1ldGhvZCI6InBhc3N3b3JkIiwidGltZXN0YW1wIjoxNzIzNzUwMDI3fV0sInNlc3Npb25faWQiOiJkMzA2OTc0Ni1iOTE5LTRkOTgtOWQ2Mi01MmRhYTZmMmIyZTgiLCJpc19hbm9ueW1vdXMiOmZhbHNlfQ.RL9gim2Zjcy8K87FLFQaNV5XQLT1AdoXxlPrg0oNpEc"
```